metadata-suggestions.md

```diff
@@ -0,0 +1,528 @@
+        "query-based": "true",
+        "file-based": "false",
+        "metadata-based": "true",
+        "_links": {
+          "entries": {
+            "href": "https://dspace7-internal.atmire.com/server/api/integration/metadata-suggestions/orcid/entries"
+          },
+          "self": {
+            "href": "https://dspace7-internal.atmire.com/server/api/integration/metadata-suggestions/orcid"
+          }
+        }
+      },
+      {
+        "id": "ciencia",
+        "name": "ciencia",
+        "type": "metadataSuggestion",
+        "query-based": "true",
+        "file-based": "false",
+        "metadata-based": "true",
+        "_links": {
+          "entries": {
+            "href": "https://dspace7-internal.atmire.com/server/api/integration/metadata-suggestions/ciencia/entries"
+          },
+          "self": {
+            "href": "https://dspace7-internal.atmire.com/server/api/integration/metadata-suggestions/ciencia"
+          }
+        }
+      },
+      {
+        "id": "ris_data_loader",
+        "name": "ris_data_loader",
+        "type": "metadataSuggestion",
+        "query-based": "false",
+        "file-based": "true",
+        "metadata-based": "false",
+        "_links": {
+          "entries": {
+            "href": "https://dspace7-internal.atmire.com/server/api/integration/metadata-suggestions/ris_data_loader/entries"
+          },
+          "self": {
+            "href": "https://dspace7-internal.atmire.com/server/api/integration/metadata-suggestions/ris_data_loader"
+          }
+        }
+      }
+    ]
+  },
+  "_links": {
+    "self": {
+      "href": "https://dspace7-internal.atmire.com/server/api/integration/metadata-suggestions"
+    }
+  },
+  "page": {
+    "size": 20,
+    "totalElements": 3,
+    "totalPages": 1,
+    "number": 0
+  }
+}
+```
+
+## Single suggestion endpoint
+**/api/integration/metadata-suggestions/<:suggestion-name>**
+
+Provide detailed information about a specific external source. The JSON response document is as follow
+```json
+{
+    "id": "pubmed",
+    "name": "pubmed",
+    "type": "metadataSuggestion",
+    "query-based": "true",
+    "file-based": "false",
+    "metadata-based": "true",
+    "_links": {
+      "entries": {
+        "href": "https://dspace7-internal.atmire.com/server/api/integration/metadata-suggestions/pubmed/entries"
+      },
+      "self": {
+        "href": "https://dspace7-internal.atmire.com/server/api/integration/metadata-suggestions/pubmed"
+      }
+    }
+}
+```
+
+Properties:
+* query-based: if the external source uses a query to suggest information to import
+* file-based: if the external source uses a file to suggest information to import
+* metadata-based: if the external source uses the current item metadata to suggest information to import
+
+Exposed links:
+* entries: the list of values managed by the external source
+
+## Linked entities
+### External source entries
+**/api/integration/metadata-suggestions/<:suggestion-name>/entries**
```

Comment:
This is one area of confusion for me. It's confusing to me to have two different query-based endpoints against the same external source (e.g. ORCID). For example, this endpoint looks really really similar to `/api/integration/externalsources/<:source-name>/entries`  : https://github.com/DSpace/Rest7Contract/blob/master/external-authority-sources.md#external-source-entries

If both of these endpoints were using ORCID, how would these calls differ? They seem like they only differ in the request parameters, and that the JSON response from either would look quite similar.  If they only differ in the request params, is it necessary to have separate endpoints, or could one endpoint just behave differently based on the params passed to it?

Reply:
The endpoints can both offer e.g. ORCID, but from a different perspective:
* The https://github.com/DSpace/Rest7Contract/blob/master/external-authority-sources.md#external-source-entries endpoint offers access for importing an ORCID record in authority control or related item
* This new endpoint offers access for importing an ORCID record in the current item

From a functional perspective, this differs because they both impact different features
From a query parameters perspective, this differs because you can make live import suggestions based on information in the current item (e.g. from files in the item, or from metadata already entered in the item). Also, the live import should suggest changes to the metadata and e.g. not suggest to add a title which is already present in the item

On top of this, you can have a setup where you use e.g. only the live import if you're not creating relationships or authority values. This would allow you to import from e.g. PubMed to enrich your publication's metadata
You can alternative have a setup where you use e.g. only the ORCID external sources for authority control, but you're not using the live import

The common part is that both features **can** search in sources which may be used in either feature and both features use the same metadata mapping. This common part is of course taken to advantage in the API implementation to avoid implementing the same class twice

Reply:
This was also clarified in this week's meeting.  Marking as resolved. https://wiki.duraspace.org/display/DSPACE/2019-11-05+DSpace+7+Entities+WG+Meeting#id-2019-11-05DSpace7EntitiesWGMeeting-Notes